compiler/z/codegen/OMRTreeEvaluator.cpp

```diff
@@ -14116,6 +14116,128 @@ bool isSettingOp(uint8_t byteValue, TR::InstOpCode::Mnemonic SI_opcode)
       return false;
    }
 
+TR::Register *
+OMR::Z::TreeEvaluator::bitpermuteEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+   {
+   TR::Node *value = node->getChild(0);
+   TR::Node *addr = node->getChild(1);
+   TR::Node *length = node->getChild(2);
+
+   auto valueReg = cg->evaluate(value);
+   auto addrReg = cg->evaluate(addr);
+
+   TR::Register *tmpReg = cg->allocateRegister(TR_GPR);
+
+   TR::Register *resultReg = cg->allocateRegister(TR_GPR);
+   // Zero result reg
+   generateRRInstruction(cg, TR::InstOpCode::getXORRegOpCode(), node, resultReg, resultReg);
+
+   if (length->getOpCode().isLoadConst())
+      {
+      // Manage the constant length case
+      int arrayLen = length->getIntegerNodeValue<int>();
```

Comment:
Try to avoid using primitive types as they have no guarantees on lower bound width. Use `int32_t` instead.

Reply:
Addressed in: 62c4af1268e0b9753c9ba29ce364b7ec782aca17